distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/AdminConfig.java

```diff
@@ -74,4 +81,34 @@ public void setLogLevels() throws IOException {
 
         logConfig.update(properties);
     }
+
+    public void updateConfig(ConfigWaitable configWaitable) throws IOException, InterruptedException {
+        Configuration config = configAdmin
+                .getConfiguration(configWaitable.getPid(), configWaitable.getLocation());
+        config.update(configWaitable.getConfigProps());
+
+        long timeoutLimit = System.currentTimeMillis() + CONFIG_UPDATE_MAX_WAIT_MILLIS;
+        while (true) {
+            if (configWaitable.isConfigured()) {
+                break;
+            } else {
+                if (System.currentTimeMillis() > timeoutLimit) {
+                    fail(String.format("Configuration change for %s not completed",
```

Comment:
This could be rephrased to include why it failed. Something like "Timed out waiting for configuration change"